api/src/main/java/com/qaengine/database/QuestionRepository.java

```diff
@@ -2,29 +2,48 @@
 
 import com.qaengine.models.Question;
 import com.qaengine.models.outputs.QuestionListElement;
+import org.springframework.data.domain.Page;
 import org.springframework.data.domain.Pageable;
 import org.springframework.data.jpa.repository.JpaRepository;
 import org.springframework.data.jpa.repository.Modifying;
 import org.springframework.data.jpa.repository.Query;
-import org.springframework.data.repository.query.Param;
+import org.springframework.data.repository.PagingAndSortingRepository;
 import org.springframework.stereotype.Repository;
 
 import javax.transaction.Transactional;
 import java.util.List;
 
 @Repository
-public interface QuestionRepository extends JpaRepository<Question, Long> {
+public interface QuestionRepository extends PagingAndSortingRepository<Question, Long> {
+
+    @Query(
+            "SELECT new com.qaengine.models.outputs.QuestionListElement(" +
+            "       q.id, q.title, q.text, q.score, q.views, q.created, q.category, " +
+            "       COUNT(a), MAX(a.created), COUNT(c), MAX(c.created)) " +
+            "FROM Question q " +
+            "LEFT JOIN Comment c ON c.question = q " +
+            "LEFT JOIN Answer a ON a.question = q " +
+            "WHERE LOWER(q.title) LIKE CONCAT('%', LOWER(:query), '%') " +
+            "GROUP BY q.id, q.title, q.text, q.score, q.created, q.category"
+    )
+    Page<QuestionListElement> listQuestions(String query, Pageable pageable);
+
     @Query(
-        "SELECT new com.qaengine.models.outputs.QuestionListElement(q.id, q.title, q.text, q.score, q.created, COUNT(c)) " +
-        "FROM Question q LEFT JOIN q.comments c " +
-        "GROUP BY q.id, q.title, q.text, q.score, q.created"
+            "SELECT new com.qaengine.models.outputs.QuestionListElement(" +
+            "       q.id, q.title, q.text, q.score, q.views, q.created, q.category, " +
+            "       COUNT(a), MAX(a.created), COUNT(c), MAX(c.created)) " +
+            "FROM Question q " +
+            "LEFT JOIN Comment c ON c.question = q " +
+            "LEFT JOIN Answer a ON a.question = q " +
+            "WHERE LOWER(q.title) LIKE CONCAT('%', LOWER(:query), '%') AND q.category.id = :categoryId " +
+            "GROUP BY q.id, q.title, q.text, q.score, q.created, q.category"
     )
-    List<QuestionListElement> listQuestions(Pageable pageable);
+    Page<QuestionListElement> listQuestionsByCategory(Long categoryId, String query, Pageable pageable);
 
     @Modifying
     @Transactional
-    @Query("update Answer a set a.accepted = false where a.questionId=:questionId")
-    void revertAnswerAccepted(@Param("questionId") Long questionId);
+    @Query("update Answer a set a.accepted = false where a.question = :question")
+    void revertAnswerAccepted(Question question);
```

Comment:
Didnt know you don't need the @Param there, cool.

Reply:
You tested it, right?

Reply:
Yep, you can use named parameters without using the annotation. 
Tested